frontend/packages/console-shared/src/utils/utils.ts

```diff
@@ -0,0 +1,31 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { getName, getNamespace } from '../selectors';
+
+export type EntityMap<A> = { [propertyName: string]: A };
+export type K8sEntityMap<A extends K8sResourceKind> = EntityMap<A>;
+
+type KeyResolver<A> = (entity: A) => string;
+
+export const getLookupId = <A extends K8sResourceKind>(entity: A): string =>
+  `${getNamespace(entity)}-${getName(entity)}`;
```

Comment:
It's not clear to me how this is used, but it's not unique across types. Is there a reason to avoid simply using the UID?

Reply:
example usage:

https://github.com/openshift/console/blob/236f5617f7fc4f3cc9b21104f72edf8bc07ba6b8/frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx#L125-L128

https://github.com/openshift/console/blob/236f5617f7fc4f3cc9b21104f72edf8bc07ba6b8/frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx#L80-L82

not sure what will be the most prevalent usage:

should I put the `getUID` function as a default to createLookup when getKey is not specified ?

Reply:
+1 on favoring UID over a composite ID that may not be unique.

Reply:
This seems like an unaddressed comment, unless I'm missing something.

Reply:
added the UID selector by default